cmd/mo-service/config.go

```diff
@@ -153,20 +156,61 @@ func (c *Config) validate() error {
 func (c *Config) createFileService(defaultName string) (*fileservice.FileServices, error) {
 	// create all services
 	services := make([]fileservice.FileService, 0, len(c.FileServices))
+
+	type perfCounterInfo struct {
+		fsName      string
+		perfCounter *perfcounter.Counter
+	}
+	perfCounterInfos := make([]perfCounterInfo, 0, len(c.FileServices))
+
 	for _, config := range c.FileServices {
 
 		// for old config compatibility
 		if strings.EqualFold(config.Name, "s3") {
 			config.Name = defines.SharedFileServiceName
 		}
 
-		service, err := fileservice.NewFileService(config)
+		perfCounter := new(perfcounter.Counter)
+		service, err := fileservice.NewFileService(config, perfCounter)
 		if err != nil {
 			return nil, err
 		}
 		services = append(services, service)
+		perfCounterInfos = append(perfCounterInfos, perfCounterInfo{
+			fsName:      service.Name(),
+			perfCounter: perfCounter,
+		})
+
 	}
 
+	// cache stats
+	//TODO add metrics exporter
+	go func() {
+		for range time.NewTicker(time.Second * 10).C {
+			for _, info := range perfCounterInfos {
+
+				reads := atomic.LoadInt64(&info.perfCounter.CacheRead)
```

Comment:
dont you need to reset counter for next ticker period?

Reply:
No, it's a global counter.
This part will be replaced with Arjun's collector.